vendor/bat-native-ads/src/bat/ads/internal/ads/serving/choose/predict_ad_embeddings.h

```diff
@@ -0,0 +1,57 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_VENDOR_BAT_NATIVE_ADS_SRC_BAT_ADS_INTERNAL_ADS_SERVING_CHOOSE_PREDICT_AD_EMBEDDINGS_H_
+#define BRAVE_VENDOR_BAT_NATIVE_ADS_SRC_BAT_ADS_INTERNAL_ADS_SERVING_CHOOSE_PREDICT_AD_EMBEDDINGS_H_
+
+#include <iostream>
+#include <vector>
+
+#include "absl/types/optional.h"
+#include "base/check_op.h"
+#include "base/notreached.h"
+#include "base/rand_util.h"
+#include "bat/ads/internal/ads/serving/choose/eligible_ads_predictor_util.h"
+#include "bat/ads/internal/ads/serving/choose/sample_ads.h"
+#include "bat/ads/internal/ads/serving/eligible_ads/pacing/pacing.h"
+#include "bat/ads/internal/ads/serving/targeting/user_model_info.h"
+#include "bat/ads/internal/base/numbers/number_util.h"
+#include "bat/ads/internal/processors/contextual/text_embedding/text_embedding_html_event_info.h"
+
+namespace ads {
+
+template <typename T>
+absl::optional<T> PredictAdEmbeddings(
```

Comment:
The name of this function does not really match its return. It returns a final selected ad. Perhaps the part that does predicting and the part that does the ad drawing/sampling should be separate functions?

Reply:
Also any functions that return absl::optional should start wiith `Maybe`

Reply:
as discussed with @ptjames changed naming to `PredictAdUsingEmbeddings`.
@tmancey do you want me to add `Maybe` to this and also to `predict_ad.h`?

Reply:
Any functions which return absl::optional should start with `Maybe`